lib/chat/chunked_files_multikey.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Chat.ChunkedFilesMultikey do
+  @moduledoc false
+
+  alias Chat.Db
+  alias Chat.Utils
+
+  @chunk_group_size 1_000_000_000
+
+  def generate(file_key, file_size, <<initial_secret::binary-size(32)>>) do
+    secrets_needed_amount = trunc(file_size / @chunk_group_size)
+
+    secrets_needed =
+      for amount <- 0..secrets_needed_amount,
+          amount > 0,
+          do: encrypt_secret(additional_secret(), initial_secret)
+
+    Db.put({:file_secrets, file_key}, [initial_secret | secrets_needed] |> Enum.join(""))
```

Comment:
Why to store initial secret ? we should not
Instead, do `secrets_needed |> Enum.join("")`